edb/pgsql/resolver/command.py

```diff
@@ -1345,6 +1363,31 @@ def is_default(e: pgast.BaseExpr) -> bool:
     )
 
 
+def _construct_ql_is_type(sub: s_objtypes.ObjectType) -> qlast.Expr:
+    """
+    For UPDATE ONLY and DELETE ONLY, we want to filter objects that are of
+    subject type and not of subject's subtypes.
+    """
+
+    # Construct QL AST equivalent to:
+    # .__type__ = <schema::Type><uuid>'subject-type-id'
+    return qlast.BinOp(
+        left=qlast.Path(
+            partial=True, steps=[qlast.Ptr(name='__type__')]
+        ),
+        op='=',
+        right=qlast.TypeCast(
+            expr=qlast.TypeCast(
+                expr=qlast.Constant.string(str(sub.id)),
+                type=qlast.TypeName(maintype=qlast.ObjectRef(name='uuid'))
+            ),
+            type=qlast.TypeName(
+                maintype=qlast.ObjectRef(module='schema', name='Type')
+            )
+        )
+    )
```

Comment:
We should do either:
- `.__type__ = (introspect [type name])`
- `.__type__.id = <uuid>'[type id]'`

Both should skip having to do an actual lookup.

Reply:
Yes, this would be much better option.

I've removed this whole chunk anyway, since I can use ONLY on value relation and filter out child types there.